### PR TITLE
Names bidding

### PR DESCRIPTION
**Pull request by sennui**
Implemented according to protocol repository PR: 
https://github.com/aeternity/protocol/pull/379

DONE:
- whole backend adjusted including fate, aevm etc for new argument in claim transaction with bid value
- aens SUITE tests all versions:
   old names & old behaviour (skips auctions),
   old names & new behaviour
   new names & new behaviour
   utf8 on all heights 
- some tests edited to escape auction behaviour at all heights. Done that by extending name to non-auction length (over 31 chars now)

NOT DONE:

- Swagger has only new interface
- HTTP integration test tests only new interface
- aecontract SUITE not finished as top level /test templates are not adjusted for extra argument in claim. Need to revert it and write new test that is skipped in pre-lima
- couple tests around failing due to not adjusted contract templates in /test 

Depends on unmerged aebytecode and sophia PRs (they are pointed in this PR rebar.config). Pre-merge commit will need to fix rebar.config and rebar.lock with rebased aebytecode and sophia PRs.

### CODE REVIEW

File: apps/aecontract/test/aecontract_SUITE.erl [hunk 1]
```diff
@@ -3472,7 +3472,9 @@ sophia_signatures_aens(Cfg) ->
     Acc             = ?call(new_account, 20000000 * aec_test_utils:min_gas_price()),
     NameAcc         = ?call(new_account, 20000000 * aec_test_utils:min_gas_price()),
     Ct              = ?call(create_contract, NameAcc, aens, {}, #{ amount => 100000 }),
-    Name1           = <<"bla.test">>,
+    LongPrefix      = <<"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx">>,
```
Comment (ThomasArts) [inline apps/aecontract/test/aecontract_SUITE.erl]: Why are we replacing the existing test with this one?

File: apps/aecore/src/aec_governance.erl [hunk 2]
```diff
@@ -12,10 +12,13 @@
...
          minimum_gas_price/1,
          name_preclaim_expiration/0,
-         name_claim_locked_fee/0,
+         name_claim_bid_increment/0,
+         name_claim_fee/1,
+         name_claim_fee_base/0,
          name_claim_max_expiration/0,
          name_protection_period/0,
-         name_claim_preclaim_delta/0,
+         name_claim_preclaim_timeout/0,
```
Comment (ThomasArts) [inline apps/aecore/src/aec_governance.erl]: could we not simply keep the name name_claim_preclaim_delta?

Comment (ThomasArts) [reply apps/aecore/src/aec_governance.erl]: it is a delta, not really a time

Comment (sennui) [reply apps/aecore/src/aec_governance.erl]: @ThomasArts I change that to correlate with name_claim_bid_timout. In both cases it is not a time range that matters but the action that happens when they expire. For such activities, I found word `timeout` more suitable. Similar to timers, which, those functions represent in some abstract form, in fact, configure.

File: apps/aecontract/test/aecontract_SUITE.erl (hunk 1 above)
Comment (sennui) [reply apps/aecontract/test/aecontract_SUITE.erl]: @ThomasArts In some cases of old test we just want to use registered name. The Long Prefix falls back to non-auction behaviour (it's over 32 characters). Surly, like that it is for state channel test. Here, I think, we might want to have new test with auction and then with short name.

**Comment by sennui** [comment]
> Would it make sense to have a first PR for the transactions... actually working for all heights and then a second PR for the contract stuff?

I don't thinks it is possible as contracts use now the same backend (primop) that I am changing for this functionality. 

Currently remaining bit is support of old contracts running on AEVM (I updated it to new stuff and it is not easy to make it work on both, no version around)

File: apps/aecore/src/aec_aens_governance.erl [hunk 3]
```diff
@@ -0,0 +1,66 @@
+%%%-------------------------------------------------------------------
+%%% @doc Placeholder for functions instrumenting Naming auctions
+%%%
+%%% @end
+%%%-------------------------------------------------------------------
+-module(aec_aens_governance).
```
Comment (ThomasArts) [inline apps/aecore/src/aec_aens_governance.erl]: I don't like the idea of spreading out governance in multiple modules.... hard to get a feeling what is under governance and what is not.

Comment (sennui) [reply apps/aecore/src/aec_aens_governance.erl]: I followed up here on what we did with coinbase inflation function. I guess it was for code clarity. Agree with you that holding everything in single governance module may be bigger hint for newcomer. I am not attached to the new module at all, if you think it's clearer to bring it to home governance, I am all up for it. Let me know.

File: apps/aens/src/aens_claim_tx.erl [hunk 4]
```diff
@@ -196,7 +234,7 @@ account_pubkey(#ns_claim_tx{account_id = AccountId}) ->
 
 -spec version(tx()) -> non_neg_integer().
 version(_) ->
```
Comment (ThomasArts) [inline apps/aens/src/aens_claim_tx.erl]: This must be depending on  height

Comment (ThomasArts) [reply apps/aens/src/aens_claim_tx.erl]: Please make all tests pass and compatible with synchronizing to an existing chain.

Comment (sennui) [reply apps/aens/src/aens_claim_tx.erl]: Done

**Pull request closed without merging**